src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -642,28 +642,12 @@ public static void Move(string sourceFileName, string destFileName)
 
         public static void Encrypt(string path)
         {
-            if (path == null)
-                throw new ArgumentNullException(nameof(path));
```

Comment:
I don't think we should be removing these argument null checks.  If you pass in a null argument, regardless of platform, it's usage error.  Plus we should be keeping as much code consolidated as possible, including argument validation.  If memory serves, we keep such ANEs even when followed by a PNSE in most other places in coreclr/corefx, e.g. memory mapped files; I don't think we should deviate from that here.
cc: @pjanotti, @JeremyKuhne